fmlloader/src/main/java/net/minecraftforge/fml/loading/RuntimeDistCleaner.java

```diff
@@ -53,8 +54,19 @@ public int processClassWithFlags(final Phase phase, final ClassNode classNode, f
         AtomicBoolean changes = new AtomicBoolean();
         if (remove(classNode.visibleAnnotations, DIST))
         {
-            LOGGER.error(DISTXFORM, "Attempted to load class {} for invalid dist {}", classNode.name, DIST);
-            throw new RuntimeException("Attempted to load class "+ classNode.name  + " for invalid dist "+ DIST);
+            final String message = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE)
+                    .walk(stream -> stream.filter(Predicate.not(it ->
+                    {
+                        final String cname = it.getClassName();
+                        return cname.startsWith("cpw.mods.cl") || cname.equals("java.lang.ClassLoader") // Exclude ModuleClassLoader and friends
+                                || cname.startsWith("cpw.mods.modlauncher") // Exclude the transformer callers
+                                || cname.equals("java.lang.Method") || cname.equals("java.lang.Class") || cname.equals("java.lang.Field") // Exclude Class#getDeclaredMethods, and any other reflection methods
+                                || cname.equals("net.minecraftforge.fml.loading.RuntimeDistCleaner"); // And finally, exclude ourselves
```

Comment:
Is there any reason why these shouldn't be grabbed from the class itself? I feel like that would be more robust as if the location changes (highly unlikely for java classes, a little less unlikely for Forge if there is ever another change).

Reply:
No real reason, it's just extremely unlikely for the jdk classes to change